modules/contact/contact.resolver.js

```diff
@@ -14,23 +17,90 @@ const contactQuery = {
 };
 
 const contactMutation = {
-  addContact: async (parent, args) =>
-    (await contactService.addContact(args.contact)) || {
-      statusCode: 400,
-      message: CONTACT_ALREADY_EXIST,
-    },
+  addContact: async (parent, args) => {
+    try {
+      if (!args.upload) {
+        return await contactService.addContact(args.contact);
+      }
 
-  deleteContact: async (parent, args) =>
-    (await contactService.deleteContact(args.id)) || {
-      statusCode: 404,
-      message: CONTACT_NOT_FOUND,
-    },
+      const uploadResult = await uploadFiles(args.upload);
+      const imageResults = await uploadResult;
 
-  updateContact: async (parent, args) =>
-    (await contactService.updateContact(args.id, args.contact)) || {
-      statusCode: 404,
-      message: CONTACT_NOT_FOUND,
-    },
+      const images = await Promise.all(imageResults).then(res => res);
```

Comment:
Are you sure that we need here `Promise.all`? Is `imageResults` an array of promises?

Reply:
Please, move all of this logic to the service, let's make resolvers as thin as possible